sljit_src/sljitNativeX86_common.c

```diff
@@ -496,35 +501,69 @@ static sljit_u32 execute_get_xcr0_low(void)
 {
 	sljit_u32 xcr0;
 
-#if defined(_MSC_VER) && _MSC_VER >= 1400
+#if (defined(_MSC_VER) && _MSC_VER >= 1400) || defined(__INTEL_COMPILER) \
+	|| (defined(__INTEL_LLVM_COMPILER) && defined(__XSAVE__))
 
 	xcr0 = (sljit_u32)_xgetbv(0);
 
-#elif defined(__GNUC__) || defined(__INTEL_COMPILER) || defined(__SUNPRO_C) || defined(__TINYC__)
+#elif defined(__TINYC__)
+
+	__asm__ (
+		"xorl %%ecx, %%ecx\n"
+		".byte 0x0f\n"
+		".byte 0x01\n"
+		".byte 0xd0\n"
+		"movl %0, %%eax\n"
+		: "+m" (xcr0)
+		:
+#if defined(SLJIT_CONFIG_X86_32) && SLJIT_CONFIG_X86_32
+		: "eax", "ecx", "edx"
+#else /* !SLJIT_CONFIG_X86_32 */
+		: "rax", "rcx", "rdx"
+#endif /* SLJIT_CONFIG_X86_32 */
+	);
+
+#elif (defined(__INTEL_LLVM_COMPILER) && __INTEL_LLVM_COMPILER < 20220100) \
```

Comment:
Why `< 20220100` is needed? Does intel supports AT&T

Reply:
that is the version that adds support for writing [multiple syntax](https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html#Multiple-assembler-dialects-in-asm-templates) that is used next, the non LLVM compiler (just like gcc) always had it AFAIK

Reply:
This code guards an AT&T syntax code, and only for older "llvm intel" compilers. Is there such thing btw? I thought intel has its own compiler. If you tested it, then it is ok.

Reply:
the original intel compiler is now obsolete and has been replaced by the "llvm intel" compiler going forward.

AT&T syntax is supported by all compilers AFAIK

Reply:
> the non LLVM compiler (just like gcc) always had it AFAIK

gcc < 3 doesn't support it (added to the list), so there might be also an ancient legacy intel compiler that would break, but considering that they are all obsolete already it should be ok